api/handlers/refererhandler.py

```diff
@@ -211,9 +189,23 @@ def delete(self, cont_name, cid, _id):
             self.abort(404, 'Analysis {} not removed from container {} {}'.format(_id, cont_name, cid))
 
 
+    def upload(self, **kwargs):
+        """Upload ad-hoc analysis outputs generated offline."""
+        _id = kwargs.get('_id')
+        analysis = self.storage.get_container(_id)
+        parent = self.storage.get_parent(analysis['parent']['type'], analysis['parent']['id'])
+        permchecker = self.get_permchecker(parent)
+        permchecker(noop)('POST')
+
+        if analysis.get('job'):
+            raise InputValidationException('Analysis created via a job does not allow file uploads')
```

Comment:
Maybe I missed it but we might want to add a check to ensure the analysis doesn't already have inputs. I believe this action should only be completed once.